### ISSUE TITLE
Hoist non-React static methods in HOC components

### ISSUE THREAD

**Issue by uncvrd** [gladly-team/next-firebase-auth#224]
**Is your feature request related to a problem? Please describe.**

Hello! I have a next.js application that requires a persistent layout between pages (for example: a sidebar on each page of a dashboard). I followed Adam Wathan's popular blog post regarding this problem [here](https://adamwathan.me/2019/10/17/persistent-layout-patterns-in-nextjs/). 

His final solution (the bonus at the very bottom) proposes to add a static property to each `Component` which assigns the specific persistent layout that wraps around the Next page. For example, you would do something like this:

```
const Login: PageWithLayoutType = () => {
    return (

    )
}

// ADD THIS LINE
Login.getLayout = getLayout;

export default Login;
```

The issue that I've encountered when using this auth library, specifically `withUserAuth` HOC, is that HOCs do not copy over static properties ([documented in the React Docs here](https://reactjs.org/docs/higher-order-components.html#static-methods-must-be-copied-over)). So in my example above, the `getLayout` static property does not get copied over. React provides a few solutions but 2 of the 3 require the user (me) to be able to modify the `HOC` component to manually export the specific static properties. There is one solution that is mentioned in the docs which I will place below...

**Describe the solution you'd like and how you'd implement it**
The react docs suggest using the [hoist-non-react-statics](https://github.com/mridgway/hoist-non-react-statics) to...you guessed it, hoist the static props that are passed from the HOC function parameters to the newly created wrapped component.

It would appear, all you'd have to add is:

```
hoistNonReactStatic(WithAuthUserHOC, ChildComponent);
// ^^ add the above line right before the return statement in `withAuthUser`
return WithAuthUserHOC
```

**Is this a breaking change?**
I'm not confident enough to provide an answer here, but I would not think it would.

**Describe alternatives you've considered**
The other alternatives are found here: https://reactjs.org/docs/higher-order-components.html#static-methods-must-be-copied-over

Would love to know if a change like this is feasible or if this is out of scope for the project. Thanks for your time!! :)

**Comment by kmjennison** [comment]
Thanks for the clear description!

Using `hoist-non-react-statics` makes sense to me, and a PR is welcome.

**Comment by uncvrd** [comment]
Awesome! I've been using a local fork with hoisted statics and haven't run in to any issues. I'll create a PR within the next few days.

**Comment by kmjennison** [comment]
Closed in #237.

**Comment by kmjennison** [comment]
Released in [v0.13.3-alpha.0](https://github.com/gladly-team/next-firebase-auth/releases/tag/v0.13.3-alpha.0).

**Comment by kmjennison** [comment]
We also have to add `hoist-non-react-statics` to the whitelisted Webpack externals. Will fix in a new PR shortly.
